src/lib/prov/tpm2/tpm2_util.h

```diff
@@ -156,6 +162,15 @@ struct esys_liberator {
 template <typename T>
 using unique_esys_ptr = std::unique_ptr<T, esys_liberator>;
 
+template <typename T>
+constexpr unique_esys_ptr<T> make_unique_esys_ptr(T obj) {
+   auto ptr = unique_esys_ptr<T>(static_cast<T*>(std::calloc(1, sizeof(T))));
```

Comment:
Note to self: This might actually need to be:
```suggestion
   auto ptr = unique_esys_ptr<T>(std::launder(reinterpret_cast<T*>(std::calloc(1, sizeof(T)))));
```

[[std::launder](https://en.cppreference.com/w/cpp/utility/launder)]